### PR TITLE
almost done

### PR DESCRIPTION
**Pull request by DeveshRamiya**

### CODE REVIEW

**Review by pjjajal — COMMENTED** [review]
Please look at the classes you actually need for MI estimation. I think it is only the MINE class and a couple of others, remove any classes you don't need.

**Comment by DeveshRamiya** [comment]
The main function which calculates the MINE utilises all the classes except D_mine (which was my own implementation of MINE, it was an experiment class I tried writing which didn’t work)

File: mutualInfo/layers.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                      kernel_size=kernel_size, stride=stride, padding=padding),
+            nn.BatchNorm2d(out_features),
+            nn.ReLU(),
+        )
+
+    def forward(self, x):
+        return self.block(x)
+
+
+class MNISTClassifier(nn.Module):
```
Comment (pjjajal) [inline mutualInfo/layers.py:40]: Where is this used?

File: mutualInfo/layers.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        h = self.layers(x)
+        return torch.softmax(self.fc1(x))
+
+    def loss_fn(self, x, y):
+        out = self.forward(x)
+        loss = self.loss(out, y)
+        return loss
+
+
+class LinearGenerator(nn.Module):
```
Comment (pjjajal) [inline mutualInfo/layers.py:62]: Where is this used?

File: mutualInfo/layers.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            nn.Linear(500, 500),
+            nn.LeakyReLU(),
+            nn.BatchNorm1d(500),
+            nn.Linear(500, output_dim))
+
+    def forward(self, x):
+        return self.layers(x)
+
+
+class LinearDiscriminator(nn.Module):
```
Comment (pjjajal) [inline mutualInfo/layers.py:79]: Where is this used?

File: mutualInfo/layers.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        self.input_dim = input_dim
+
+    def forward(self, x):
+        if(len(x.shape) != 2):
+            x = x.view(x.shape[0], -1)
+
+        return self.layers(x)
+
+
+class DCGanGenerator(nn.Module):
```
Comment (pjjajal) [inline mutualInfo/layers.py:101]: Where is this used?

**Review by pjjajal — COMMENTED** [review]
There are quite a few classes in layers.py that are not used.

**Comment by DeveshRamiya** [comment]
I see, sorry, I didn’t realise.

**Pull request closed without merging**